### DISCUSSION TITLE
Energy Probe and Meter Box

### DISCUSSION THREAD

**Discussion by Morphyloon** [TechReborn/TechReborn#2537] [Ideas]
As it is now, there is no way to tell what, if any, energy is stored or moving within the system of energy generating blocks (solar, wind, etc), energy carrying blocks (cables), or the blocks that redistribute energy (batterybox, MSU, etc). The only information we have is observing the energy reserve meter within machines and batteries.

I'd like to propose something to let the player know more. A tool, similar to the wrench, that you poke at a block to give you information. This is assuming such information can be read at all such as the reserve meter of a cable, maybe even the amount of power going in/out of a cable/MSU?

An advanced option would be a new block. A meter box that tells you how much energy is going through it in real time. It could be further enhanced by allowing energy use to be monitored over time. The readout would probably have to be a fixed period to keep things from becoming too complicated. Another enhancement might be a redstone signal generated when the energy drops to (near) zero-- and/or the front texture face changes from a green light to a red light. The GUI for the Meter block would have to contend with how to handle energy flow in or out of each face (default to all inputs when created?), otherwise it would need a fixed setting like left/top for input and right/bottom for output but this would limit it's placement in certain cabling designs.

**Comment by imcasper** [comment]
I support it too. This is a very good idea. Now it is possible to estimate the produced / consumed energy only indirectly.
I would suggest that you first implement the option that is easier and faster:
- tool "voltmeter"
or
- block "energy-meter"

**Comment by imcasper** [comment]
After adding some mod
- megane
- wthit
- or something other
... When you move the cursor over the wire, the energy is shown.
Thus, a separate voltmeter is unnecessary.
My experience with "immersive engineering" mod shows that running with a voltmeter is fun at first, but uncomfortable, and you constantly forget it at home.
